src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private StorageDao storageDao;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Override
+    public Map<String, Integer> process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            operationStrategy.get(transaction.getType()).performOperation(transaction);
```

Comment:
Consider adding a null check after retrieving the operation handler from the `operationStrategy`. If the handler is null, it might indicate an unsupported operation type, and you should handle this case to avoid potential `NullPointerException`.